app/src/main/java/de/tum/in/tumcampusapp/component/ui/ticket/repository/TicketsRemoteRepository.kt

```diff
@@ -0,0 +1,38 @@
+package de.tum.`in`.tumcampusapp.component.ui.ticket.repository
+
+import android.content.Context
+import de.tum.`in`.tumcampusapp.api.app.TUMCabeClient
+import de.tum.`in`.tumcampusapp.api.app.exception.NoPrivateKey
+import de.tum.`in`.tumcampusapp.component.ui.ticket.model.Ticket
+import de.tum.`in`.tumcampusapp.component.ui.ticket.model.TicketType
+import io.reactivex.Completable
+import io.reactivex.Observable
+import retrofit2.Call
+import javax.inject.Inject
+
+class TicketsRemoteRepository @Inject constructor(
+        private val context: Context,
+        private val tumCabeClient: TUMCabeClient,
+        private val ticketsLocalRepository: TicketsLocalRepository
+) {
+
+    fun fetchTickets(): Observable<List<Ticket>> = tumCabeClient.fetchTickets(context)
+
+    @Throws(NoPrivateKey::class)
+    fun fetchTicket(ticketId: Int): Call<Ticket> {
+        return tumCabeClient.fetchTicket(context, ticketId)
+    }
+
+    fun fetchTicketTypesForTickets(tickets: List<Ticket>): Completable {
+        val sources = tickets.map { fetchTicketTypesForEvent(it.eventId) }
```

Comment:
this is probably also not a good idea to do an API request for each ticket instead of one API call for the whole list...

Reply:
It's not critical, but probably add some kind of reminder to switch to a more efficient API